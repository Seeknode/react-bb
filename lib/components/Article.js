import React from 'react';

class Article extends React.Component {
  render() {
    const { article, author } = this.props;
    return (
        <div>
          <div>{article.title}</div>
          <div>{article.date}</div>
          <div>{article.firnName} {author.lastName}</div>
          <div>{article.body}</div>
        </div>
    );
  }
}

export default Article;