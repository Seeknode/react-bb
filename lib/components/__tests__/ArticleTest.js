import React from 'react';
import Article from '../Article';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const testProps = {
    article: {},
    actions: {
      lookupAuthor: jest.fn(() => ({}))
    }
  };
  const tree = renderer.create(
    <Article
      {...testProps}
    />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});