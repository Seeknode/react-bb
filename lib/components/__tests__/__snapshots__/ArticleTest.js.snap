// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  />
  <div>
    Invalid Date
  </div>
  <div>
     
  </div>
  <div
    style={
      Object {
        "paddingLeft": 10,
      }
    }
  />
</div>
`;
